### PR TITLE
[4.0][webservice][com_users] add more query param to GET

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes 

add more query parameters

### Testing Instructions
`GET http://localhost/<yoursite>/api/v1/users?state=1&group_id=2`

### Expected result
you can get result filtered for state and groupid

### Actual result
you can get all users or 1 user

### Documentation Changes Required

yes

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ae89e88f732151fd3737eba8589df489743fdb07<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27138">issues.joomla.org/tracker/joomla-cms/27138</a>.</sub>

File: api/components/com_users/Controller/UsersController.php [hunk 1]
```diff
@@ -65,4 +65,23 @@ protected function save($recordKey = null)
...
+	 * @return  static  A \JControllerLegacy object to support chaining.
+	 *
+	 * @since   4.0.0
+	 */
+	public function displayList()
+	{
+		$this->input->set('model_state',
+			[
+				'filter.state'    => $this->input->get('state', $this->input->post->get('state')),
+				'filter.group_id' => $this->input->get('groupid', $this->input->post->get('groupid')),
```
Comment (Quy) [inline api/components/com_users/Controller/UsersController.php]: Let's change `groupid` to `group_id` to be consistent with `client_id` syntax.

Comment (alikon) [reply api/components/com_users/Controller/UsersController.php]: done

File: api/components/com_users/Controller/UsersController.php [hunk 2]
```diff
@@ -65,4 +65,23 @@ protected function save($recordKey = null)
...
+	 *
+	 * @return  static  A \JControllerLegacy object to support chaining.
+	 *
+	 * @since   4.0.0
+	 */
+	public function displayList()
+	{
+		$this->input->set('model_state',
+			[
+				'filter.state'    => $this->input->get('state', $this->input->post->get('state')),
```
Comment (wilsonge) [inline api/components/com_users/Controller/UsersController.php]: We don't accept post requests here. DisplayList is a GET endpoint only in rest. Please use `$this->input->get->get('state')`

Comment (wilsonge) [reply api/components/com_users/Controller/UsersController.php]: Same goes for this entire group of PRs please

Comment (alikon) [reply api/components/com_users/Controller/UsersController.php]: if you can check if this is ok now, i will do same for the other pr's

**Comment by wilsonge** [comment]
@alikon I’m sorry are you able to do this again with the new approach to modelState. This looks much better! Sorry for not replying

**Comment by alikon** [comment]
no problem
i'll redo it with the new approach when i'll have some more spare time
i hope someonelse will be more faster than me.... 
[silly joke]
 maybe some professional payed folks  can do it

**Pull request closed without merging**